lib/logstash/outputs/loggly.rb

```diff
@@ -117,17 +134,48 @@ def send_event(url, message)
       http.verify_mode = OpenSSL::SSL::VERIFY_NONE
     end
 
-    # post message
     request = Net::HTTP::Post.new(url.path)
     request.body = message
-    response = http.request(request)
 
-    if response.is_a?(Net::HTTPSuccess)
-      @logger.info("Event send to Loggly OK!")
-    else
-      @logger.warn("HTTP error", :error => response.error!)
+    # Variable for count total retries
+    totalRetries = 0
+
+    #try posting once when can_retry is false
+    if @can_retry == false
+      @retry_count = 1
     end
+
+    
+    @retry_count.times do
+    begin
+      response = http.request(request)	
+      case response.code
+	    when HTTP_SUCCESS					# HTTP_SUCCESS :Code 2xx
+	      puts "Event send to Loggly"
+	    when HTTP_FOBIDDEN					# HTTP_FOBIDDEN :Code 403
+	      @logger.warn("User does not have privileges to execute the action.")
+	    when HTTP_NOT_FOUND					# HTTP_NOT_FOUND :Code 404
+	      @logger.warn("Invalid URL. Please check URL should be http://logs-01.loggly.com/inputs/CUSTOMER_TOKEN/tag/logstash")
+	    when HTTP_INTERNAL_SERVER_ERROR			# HTTP_INTERNAL_SERVER_ERROR :Code 500
+	      @logger.warn("Internal Server Error")
+	    when HTTP_GATEWAY_TIMEOUT				# HTTP_GATEWAY_TIMEOUT :Code 504
+	      @logger.warn("Gateway Time Out")
+	    else
+	      @logger.error("Unexpected response code", :code => response.code)
+      end # case
+
+    if [HTTP_SUCCESS,HTTP_FOBIDDEN,HTTP_NOT_FOUND].include?(response.code)	# break the retries loop for the specified response code
+      break
+    end
+    rescue StandardError => e
+      @logger.error("An unexpected error occurred", :exception => e.class.name, :error => e.to_s, :backtrace => e.backtrace)
+    end # rescue
+     
+    if totalRetries < @retry_count && totalRetries > 0
+      puts "Waiting for five seconds before retry..."
```

Comment:
This should probably use `@logger.info` otherwise it won't end up in the user's logs